packages/fuels/tests/contracts.rs

```diff
@@ -645,3 +645,70 @@ async fn contract_call_futures_implement_send() -> Result<(), Error> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_contract_set_estimation() -> Result<(), Error> {
+    setup_contract_test!(
+        foo_contract_instance,
+        wallet,
+        "packages/fuels/tests/contracts/foo_contract"
+    );
+    let foo_contract_id = foo_contract_instance.get_contract_id();
+
+    let res = foo_contract_instance.methods().foo(true).call().await?;
+    assert!(!res.value);
+
+    setup_contract_test!(
+        foo_caller_contract_instance,
+        None,
+        "packages/fuels/tests/contracts/foo_caller_contract"
+    );
+
+    let bits = *foo_contract_id.hash();
+    let res = foo_caller_contract_instance
+        .methods()
+        .call_foo_contract(Bits256(bits), true)
+        .estimate_tx_dependencies(None)
+        .await?
+        .call()
+        .await?;
+
+    assert!(res.value);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_output_variable_contract_id_estimation_multicall() -> Result<(), Error> {
+    setup_contract_test!(
+        foo_contract_instance,
+        wallet,
+        "packages/fuels/tests/contracts/foo_contract"
+    );
+
+    let foo_contract_id = foo_contract_instance.get_contract_id();
+
+    setup_contract_test!(
+        foo_caller_contract_instance,
+        None,
+        "packages/fuels/tests/contracts/foo_caller_contract"
+    );
+
+    let bits = *foo_contract_id.hash();
+    let contract_methods = foo_caller_contract_instance.methods();
+
+    let mut multi_call_handler = MultiContractCallHandler::new(wallet.clone());
+    multi_call_handler.tx_params(Default::default());
+
+    (0..3).for_each(|_| {
+        let call_handler = contract_methods.call_foo_contract(Bits256(bits), true);
+        multi_call_handler.add_call(call_handler);
+    });
```

Comment:
I would add three different `call_handlers` with different contract calls just in case. Maybe even mix in one that doesn't use a contract as an input.

Reply:
I would have to add two more `sway` files that would do the same. Maybe I shouldn't do that, but in the multicall test I added a call that doesn't need a contract id